test/e2e/util.go

```diff
@@ -0,0 +1,448 @@
+//+build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"reflect"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1alpha1 "github.com/openshift/external-dns-operator/api/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/aws-sdk-go/service/route53"
+)
+
+var defaultExtDNSTemplate = operatorv1alpha1.ExternalDNS{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "test-extdns",
+		Namespace: "publish-external-dns",
+	},
+	Spec: operatorv1alpha1.ExternalDNSSpec{
+		Source: operatorv1alpha1.ExternalDNSSource{
+			ExternalDNSSourceUnion: operatorv1alpha1.ExternalDNSSourceUnion{
+				Type: operatorv1alpha1.SourceTypeService,
+				Service: &operatorv1alpha1.ExternalDNSServiceSourceOptions{
+					ServiceType: []corev1.ServiceType{
+						corev1.ServiceTypeLoadBalancer,
+						corev1.ServiceTypeClusterIP,
+					},
+				},
+				Namespace: StringRef("publish-external-dns"),
```

Comment:
We thought of removing the namespace filter from the API. Better to remove it, this is not going to change the rest of the tests.